x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/rules/api/util/with_existing_migration_id.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RequestHandler, RouteMethod } from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import type { SecuritySolutionRequestHandlerContext } from '../../../../../types';
+import type { RuleMigration } from '../../../../../../common/siem_migrations/model/rule_migration.gen';
+
+export const MIGRATION_ID_NOT_FOUND = (id: string) =>
+  i18n.translate('xpack.securitySolution.api.migrationIdNotFound', {
+    defaultMessage: `No Migration found with id: {id}`,
+    values: {
+      id,
+    },
+  });
+
+type WithMigration<T> = T & { migration: RuleMigration };
+
+/**
+ * Checks the existence of a valid migration before proceeding with the request.
+ *
+ * if not found, it returns a 404 error with a message.
+ * if found, it adds the migration to the context.
+ *
+ * */
+export const withExistingMigration = <
+  P extends { migration_id: string },
+  Q = unknown,
+  B = unknown,
+  Method extends RouteMethod = never
+>(
+  handler: RequestHandler<P, Q, B, WithMigration<SecuritySolutionRequestHandlerContext>, Method>
+): RequestHandler<P, Q, B, SecuritySolutionRequestHandlerContext, Method> => {
+  return async (context, req, res) => {
+    const { migration_id: migrationId } = req.params;
+    const ctx = await context.resolve(['securitySolution']);
+    const ruleMigrationsClient = ctx.securitySolution.getSiemRuleMigrationsClient();
+    const storedMigration = await ruleMigrationsClient.data.migrations.get({
+      id: migrationId,
+    });
+
+    if (!storedMigration) {
+      return res.notFound({
+        body: MIGRATION_ID_NOT_FOUND(migrationId),
+      });
+    }
+
+    Object.defineProperty(context, 'migration', {
+      value: storedMigration,
+    });
```

Comment:
Is this ever used? I can not find it

Reply:
Also, it is a bit hacky assigning this to the `context` object this way. Can't we pass the `migration` as the 4th parameter to the `handler`?

Reply:
Actually it is a common practice for middlewares to mutate `request` object. But i did not want to mutate request object since i am not sure how kibana handles it.  Therefore, i decided to mutate `Context` object since it has only `securitysolution` related data.

It looks like we do not have a standard for it as of yet. I am okay adding a new parameter as well.


> Is this ever used? I can not find it

It will be used when `name` of the migration is added and is returned as a part of `stats` endpoint. This just lays the foundation.

Reply:
> i decided to mutate Context object since it has only securitysolution related data.

At this point, the context is not yet resolved, so we are mutating the raw context object provided by Kibana:
```ts
{
  resolve,
  core,
  securitySolution,
  // ... the rest of the entries registered, they are all async functions to retrieve the services
  migration, // <- the migration document injected 
}
``` 
This is a bit unconventional, also, if someday we have a `migration` service at top level, this would be silently overriding it.

> It will be used when name of the migration is added and is returned as a part of stats endpoint. This just lays the foundation.

Okay, then could we postpone adding this logic, injecting the `migration` document object, until it is used?

Reply:
For now i have removed the injection of `migration` object but i think we will have decided how to go about it. Let's talk in today's meeting.

See https://github.com/elastic/kibana/pull/219597/commits/02c4c6e603a7164efe256101a295c3b3b56b6df2